eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -216,7 +217,19 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
 
   private def gotoAbortedOrStop(d: PaymentAborted): State = {
     if (d.pending.isEmpty) {
-      myStop(d.request.replyTo, Left(PaymentFailed(id, paymentHash, d.failures)))
+      val pf = PaymentFailed(id, paymentHash, d.failures)
+      if (cfg.storeInDb) {
+        // In cases where we fail early (router error during the first attempt), the DB won't have an entry for that
+        // payment, which may be confusing for users.
+        nodeParams.db.payments.listOutgoingPayments(cfg.parentId) match {
```

Comment:
Writing to db here forces you to call `db.payments.listOutgoingPayments` in order to tell whether there were child payments. Why don't you do it earlier, when you have that information at hand?

```scala
when(WAIT_FOR_ROUTES) {
  ...
  case Event(Status.Failure(t), d: PaymentProgress) =>
    if (d.pending.isEmpty && d.failures.isEmpty) {
    // no child payment was ever attempted, write to db here
    }
  ..
}
```

Reply:
Good idea, done!